persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -294,10 +297,11 @@ private void waitForSchema(int delayMillis) {
 
   private static ClientState clientStateForExternalCalls(@Nonnull ClientInfo clientInfo) {
     if (clientInfo.publicAddress().isPresent()) {
-      new ClientStateWithPublicAddress(
+      return new ClientStateWithPublicAddress(
           clientInfo.remoteAddress(), clientInfo.publicAddress().get());
     }
-    return ClientState.forExternalCalls(clientInfo.remoteAddress(), null);
+    return new ClientStateWithPublicAddress(
```

Comment:
I'd prefer we fix downstream handling of `ClientState`. Something like? Or we could just return empty tables.

```diff
diff --git a/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java b/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java
index ec63191..71a0ebe 100644
--- a/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java
+++ b/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java
@@ -120,15 +120,14 @@ public class ProxyProtocolQueryInterceptor implements QueryInterceptor {
       QueryOptions options,
       Map<String, ByteBuffer> customPayload,
       long queryStartNanoTime) {
-    if (!isSystemLocalOrPeers(statement)) {
+    ClientState clientState = state.getClientState();
+    if (!isSystemLocalOrPeers(statement) || !(clientState instanceof ClientStateWithPublicAddress)) {
       return null;
     }
     SelectStatement selectStatement = (SelectStatement) statement;
     List<List<ByteBuffer>> rows;
-    ClientState clientState = state.getClientState();
-    assert clientState instanceof ClientStateWithPublicAddress;
     InetSocketAddress publicAddress = ((ClientStateWithPublicAddress) clientState).publicAddress();
     if (publicAddress == null) {
```